Backend/WebApi-PointMap/ErrorHandling/DatabaseErrorHandler.cs

```diff
@@ -0,0 +1,31 @@
+﻿using DataAccessLayer.Database;
+using System;
+using System.Data.Entity;
+using System.Data.Entity.Infrastructure;
+using System.Data.Entity.Validation;
+using System.Net.Http;
+using System.Net;
+using static ServiceLayer.Services.ExceptionService;
+
+namespace WebApi_PointMap.ErrorHandling
+{
+    public class DatabaseErrorHandler
+    {
+        public static HttpResponseMessage HandleException(Exception e, DatabaseContext _db)
```

Comment:
When is this called exactly? After a .savechanges() fails?

Reply:
This is the order of the error handling calls: SomeErrorHandler(specific to the controller and only if needed) -> DatabaseErrorHandler -> HttpErrorHandler -> AuthorizationErrorHandler -> GeneralErrorHandler. The last four are always called in the above order if an exception occurs.